### DISCUSSION TITLE
Determining distance reconstructed object to detector

### DISCUSSION THREAD

**Discussion by MartijnD-1** [astra-toolbox/astra-toolbox#462] [Q&A] [answered]
Dear all,

I have been using ASTRA toolbox for some while and it's a great piece of software. We are going to use it in an upcoming clinical trial. 

For a scatter correction task I require two specific measurements. When I have a reconstructed object I will apply a threshold on the volume so that get a binary volume. Then, for every detector pixel, I need to know two points: the point in which the ray enters the binary object and the point in which its exists.

Of course I can manage a raycasting solution for this problem myself, but ideally I would like to stay in the ASTRA toolbox environment since my detector geometry is quite complex. Would you have any idea how to achieve this within ASTRA? Thanks a lot!

**Comment by askorikov** [comment, marked as answer]
Hi! I think getting intersection coordinates will be quite tricky to do with Astra unfortunately. Just to confirm, it would not be enough to know just the projected thickness of the object?

**Reply by MartijnD-1** [reply]
Thanks Alexander, that's what I thought as well. I'll make something outside of Astra 👍
